src/components/Taxonomy/OptGroupVersionSelector.tsx

```diff
@@ -38,25 +69,7 @@ const OptGroupVersionSelector = ({
     name: version.name,
     type: version.versionType,
   }));
-
-  const { published, beta, archived } = options.reduce<OptGroups>(
-    (acc, curr) => {
-      const type = curr.type.toLowerCase() as Lowercase<VersionStatusType>;
-      acc[type].push(curr);
-      return acc;
-    },
-    {
-      published: [],
-      beta: [],
-      archived: [],
-    },
-  );
-
-  const optGroups = [
-    { label: t('taxonomyVersions.status.PUBLISHED'), options: published },
-    { label: t('taxonomyVersions.status.BETA'), options: beta },
-    { label: t('taxonomyVersions.status.ARCHIVED'), options: archived },
-  ].filter(group => group.options.length > 0);
+  const optGroups = useMemo(() => generateOptionGroups(options, t), [options, t]);
```

Comment:
Når du bruker variabler som ikke er primitive må du sørge for at de også er memoized. I dette tilfellet er `options` et array og ikke memoized. Da vil optGroups kalkuleres hver render uansett.